base/broadcast.jl

```diff
@@ -368,35 +368,35 @@ end
 # additional definitions for Nullable
 # fast one & two-argument implementations
 function broadcast{T}(f, x::Nullable{T})
-    restype = promote_op(f, T)
+    S = promote_op(f, T)
     if x.isnull
-        Nullable{restype}()
+        Nullable{S}()
     else
-        Nullable{restype}(f(x.value))
+        Nullable{S}(f(x.value))
     end
 end
 
 function broadcast(f, x::Union{Nullable, Number}, y::Union{Nullable, Number})
     if isa(x, Number) && isa(y, Number)
         throw(ArgumentError("broadcast is not supported on numbers"))
     end
-    restype = promote_op(f, eltype(x), eltype(y))
-    if isnullvalue(x) | isnullvalue(y)
-        Nullable{restype}()
+    S = promote_op(f, eltype(x), eltype(y))
+    if (isa(x, Nullable) && x.isnull) | (isa(y, Nullable) && y.isnull)
```

Comment:
This is intentional. It avoids a branch (or at least it did before I inlined the `isnullvalue`; haven't looked at new generated code).